src/test/HST/CoreAlgorithmTests.hs

```diff
@@ -10,7 +10,9 @@ import           Test.Hspec                     ( Spec
                                                 )
 import           Test.HUnit.Base                ( assertFailure )
 
-import           HST.CoreAlgorithm
+import qualified HST.CoreAlgorithm             as CA
```

Comment:
I don't think that we need to import this module `qualified`. An explicit import of `compareCons` would be sufficient.